api/modules/auth/auth.controller.js

```diff
@@ -18,59 +18,103 @@ limitations under the License.
 const jwt = require('jsonwebtoken');
 const env = require('dotenv').config();
 const { default: to } = require('await-to-js');
+const passport = require('passport');
 
 // Constant Imports
 const authConstants = require('./auth.constants');
 const globalConstants = require('../../constants/constants.js');
 
+// Helper Functions imports
+const { sendPasswordResetMail, createJwtPayload } = require('./auth.helper');
+
 // Model imports
 const User = require('./auth.model');
 
 // Logger import
 const logger = require('../../config/logger/logger');
-const { sendPasswordResetMail } = require('./auth.helper');
 
 /**
- * Created login controller to send response to the user on login
+ * Controller Function to login the user
  * @param {Object} req
  * @param {Object} res
- * @return {Object}
+ * @param {callback} next
  */
-const login = async (req, res) => {
-  if (req.user) {
-    logger.info('User is logged in successfully (loginController)', { userId: req.user._id });
-    return res.status(200).json({ success: 1, message: globalConstants.responseMessages.logInUser.success, data: req.user });
-  } else {
-    logger.error('No user in req. login Problem (loginController)', { req: req });
-    return res.status(200).json({ success: 0, message: globalConstants.responseMessages.logInUser.failure, data: null });
-  }
+const login = (req, res) => {
+  passport.authenticate('login', (err, user, info) => {
+    if (err) {
+      logger.error('Passport login Strategy returned error (loginController)', { error: err }, { info: info }, { user: user });
+      return res.status(409).json({ success: 0, message: info?.message, data: { err } });
```

Comment:
please create separate file for status codes.

Reply:
Done